kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -191,6 +212,72 @@ private void handleMetadata(ColumnVector metadataVector) {
     }
   }
 
+  /**
+   * Checks whether each of the current transaction's {@link DomainMetadata} conflicts with the
+   * winning transaction at any domain.
+   *
+   * <ol>
+   *   1. Accept the current transaction if its set of metadata domains does not overlap with the
+   *   winning transaction's set of metadata domains.
+   *   <p>2. Otherwise, fail the current transaction unless each conflicting domain is associated
+   *   with a domain-specific way of resolving the conflict.
+   * </ol>
+   *
+   * @param actionBatch action batch from the winning transactions
+   */
+  private void handleDomainMetadata(ColumnarBatch actionBatch) {
+    // Extract the domain metadata map from the winning transactions. Allow a DM action in a later
+    // transaction to overwrite the DM in an earlier transaction for the same domain.
+    Map<String, DomainMetadata> winningDomainMetadataMap =
+        DomainMetadataUtils.extractDomainMetadataMap(
+            inMemoryIterable(actionBatch.getRows()), CONFLICT_RESOLUTION_SCHEMA, true);
```

Comment:
I reviewed this again, and we shouldn’t allow duplicates here.

Initially, I was thinking multiple winning transaction logs might be processed together and returned in a single batch, so we should keep the latest entry for each domain. I’ve just confirmed each action batch comes from a single log/checkpoint file, so duplicates aren’t actually expected and should never happen.

I’ve removed the `allowDuplicate` flag in `extractDomainMetadataMap`, so it will now always raise an exception for duplicates. Thanks for flagging this!